presto-main/src/test/java/io/prestosql/execution/MockResourceUpdatingQueryExecution.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution;
+
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import io.prestosql.Session;
+import io.prestosql.server.BasicQueryInfo;
+import io.prestosql.spi.ErrorCode;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.units.DataSize.Unit.BYTE;
+import static io.prestosql.execution.QueryState.FAILED;
+import static io.prestosql.execution.QueryState.FINISHED;
+import static io.prestosql.execution.QueryState.QUEUED;
+import static io.prestosql.execution.QueryState.RUNNING;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class MockResourceUpdatingQueryExecution
```

Comment:
Perhaps MockQueryExecutionWithResources?

Reply:
Trying to see if we can merge the two mock executions as suggested by Dain.